Aplib.Core/Belief/Beliefs/SampledMemoryBelief.cs

```diff
@@ -46,70 +46,109 @@ private int ModuloCounter
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="SampledMemoryBelief{TReference, TObservation}"/> class with an object reference,
-        /// and a function to generate/update the observation using the object reference.
-        /// This belief also stores a limited amount of previous observation samples in memory.
+        /// Initializes a new instance of the <see cref="SampledMemoryBelief{TReference, TObservation}"/> class with an
+        /// object reference, a function to generate/update the observation using the object reference, and a condition
+        /// on when the observation should be updated. This belief also stores a limited amount of previous observation
+        /// samples in memory.
         /// Optionally, the belief can always store the most recent observation, regardless of the sample rate.
         /// </summary>
+        /// <param name="metadata">
+        /// Metadata about this goal, used to quickly display the goal in several contexts.
+        /// </param>
         /// <param name="reference">
         /// The reference used to generate/update the observation. This <i>must</i> be a reference type.
         /// </param>
-        /// <param name="getObservationFromReference">A function that takes a reference and generates/updates a observation.</param>
+        /// <param name="getObservationFromReference">
+        /// A function that takes a reference and generates/updates a observation.
```

Comment:
Nit (also, not your fault)
```suggestion
        /// A function that takes a reference and generates/updates an observation.
```